### ISSUE TITLE
Access control dialog does not show the submitUrl

### ISSUE THREAD

**Issue by Phidica** [keepassxreboot/keepassxc#2788]
When requesting a login using the KeePassXC-Browser protocol (ie, formatting up JSON strings in any external program and communicating over the `kpxc_server` socket directly), if a submitUrl is provided then it is *not* shown in the access control dialogue popup which asks the user if they wish to allow that source access to a certain entry.

For a visual illustration:
```
Unencrypted message:
{
	"action": "get-logins",
	"url": "https://target.com",
	"submitUrl": "https://requester.com",
	"keys": [
		{
			"id": <connected_id>,
			"key": <connected_key>
		},
		...
	]
}
Request:
{
	"action": "get-logins",
	"message": "<encrypted message>",
	"nonce": "<snip>",
	"clientID": "<clientID>"
}
```
![Screenshot from 2019-03-13 00-31-02](https://user-images.githubusercontent.com/13155128/54206385-1ed82180-4528-11e9-9006-f030569782e6.png)

If the user chooses "Allow" and "Remember this decision" then the submitUrl is saved to the list of allowed domains in the entry's attributes.

![Screenshot from 2019-03-13 00-47-55](https://user-images.githubusercontent.com/13155128/54207226-b4c07c00-4529-11e9-99fd-e035de95bc0c.png)

## Steps to Reproduce
Send a `get-logins` request with a message that has different "url" and "submitUrl" fields and allow the request remembering the decision.

## Expected Behavior
For the user's benefit, I expected that the dialogue box would read:
```
requester.com has requested access to passwords for the following item(s). 
Please select whether you want to allow access.
    target.com - user
```

## Current Behavior
The current behaviour is that the submitUrl is *never* shown to the user, but it is ultimately saved as an allowed domain.

## Possible Solution
When a submitUrl is provided, show it in the access control dialogue so the user can confirm it is a trusted source to have access to the lookup URL.

## Context
I understand that this scenario is highly constructed, because the KeePassXC-Browser extension handles its requests to the database in such a way that the submitUrl should never significantly differ from the lookup URL and this is unlikely or impossible to be an attack vector over the web.

However, this protocol can be leveraged by any external program on the user's computer, and I believe the current behaviour of not revealing the submitUrl is both obstructing for external programs attempting to make their actions transparent to the user, and potentially dangerous as those external programs could at the same time allow a specific web domain future access to an entry without the user's explicit knowledge.

**Comment by varjolintu** [comment]
I agree that if the submitUrl differs from the main URL it should be shown.

**Comment by droidmonkey** [comment]
This was merged into 2.5.0
